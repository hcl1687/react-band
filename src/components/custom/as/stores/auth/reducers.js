export default {
  login: {
    next (state, action) {
      return {
        ...state,
        AUTH: action.payload
      }
    },
    throw (state, action) {
      // handle error
      return {
        ...state
      }
    }
  },
  logout: {
    next (state, action) {
      return {
        ...state,
        AUTH: {}
      }
    }
  }
}
