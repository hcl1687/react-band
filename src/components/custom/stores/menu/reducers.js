export default {
  showMenu: {
    next (state, action) {
      return {
        ...state,
        MENU_STATUS: action.payload
      }
    }
  },
  enableMenu: {
    next (state, action) {
      return {
        ...state,
        MENU_ENABLED: action.payload
      }
    }
  }
}
