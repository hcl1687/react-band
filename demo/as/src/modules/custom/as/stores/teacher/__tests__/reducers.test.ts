import reducers from '../reducers'

describe('custom/as/stores/teacher/reducers', () => {
  it('should render correctly', () => {
    expect(reducers).toMatchSnapshot()
  })

  it('check getTeacher', () => {
    const { getTeacher: { next: nextFun, throw: throwFun } } = reducers
    expect(nextFun({}, {
      type: 'getTeacher',
      payload: {
        id: 1,
        name: 'test'
      }
    })).toEqual({
      teacher: {
        id: 1,
        name: 'test'
      }
    })

    expect(throwFun({
      teacher: {}
    }, {
      type: 'getTeacher'
    })).toEqual({
      teacher: {}
    })
  })

  it('check editTeacher', () => {
    const { editTeacher: { next: nextFun, throw: throwFun } } = reducers
    expect(nextFun({}, {
      type: 'editTeacher',
      payload: {
        id: 1,
        name: 'test'
      }
    })).toEqual({
      teacher: {
        id: 1,
        name: 'test'
      }
    })

    expect(throwFun({
      teacher: {}
    }, {
      type: 'editTeacher'
    })).toEqual({
      teacher: {}
    })
  })
})
