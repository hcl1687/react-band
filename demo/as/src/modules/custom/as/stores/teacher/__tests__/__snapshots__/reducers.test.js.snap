// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom/as/stores/teacher/reducers should render correctly 1`] = `
Object {
  "editTeacher": Object {
    "next": [Function],
    "throw": [Function],
  },
  "getTeacher": Object {
    "next": [Function],
    "throw": [Function],
  },
}
`;
